xgboost/src/main/java/hivemall/xgboost/XGBoostUDTF.java

```diff
@@ -269,44 +274,33 @@ public void checkTargetValue(double target) throws HiveException {}
     public void process(Object[] args) throws HiveException {
         if (args[0] != null) {
             // TODO: Need to support dense inputs
-            final List<String> features = (List<String>) featureListOI.getList(args[0]);
+            final List<?> features = (List<?>) featureListOI.getList(args[0]);
+            final String[] fv = new String[features.size()];
+            for (int i = 0; i < features.size(); i++) {
+                fv[i] = (String) featureElemOI.getPrimitiveJavaObject(features.get(i));
+            }
             double target = PrimitiveObjectInspectorUtils.getDouble(args[1], this.targetOI);
             checkTargetValue(target);
-            final LabeledPoint point = XGBoostUtils.parseFeatures(target, features);
+            final LabeledPoint point = XGBoostUtils.parseFeatures(target, fv);
             if (point != null) {
                 this.featuresList.add(point);
             }
         }
     }
 
-    /**
-     * Need to override this for a Spark wrapper because `MapredContext` does not work in there.
-     */
-    protected String generateUniqueModelId() {
-        return "xgbmodel-" + String.valueOf(HadoopUtils.getTaskId());
+    private String generateUniqueModelId() {
+        return "xgbmodel-" + taskId + "-" + RandomUtils.getUUID() + "-" + sequence++;
```

Comment:
Ah, multiple models are created in a single task. So, `sequence` is required.

`taskId` could be useful just in case UUIDs are corrupting. So, okey as it is. 

So, `"xgbmodel-" + taskId + "-" + HadoopUtils.getUniqueTaskIdString() + "-" + sequence++;"` ?